src/main/webapp/rolesManagement/RolesManagement.js

```diff
@@ -0,0 +1,182 @@
+      return prev;
+    });
+  };
+
+  closeModal = () => () => {
+    this.setState(prev => {
+      prev.editRoleDialog = false;
+      return prev;
+    });
+  };
+
+  submit = (userUpdated) => {
+    this.setState(prev => {
+      prev.editRoleDialog = false;
+      prev.users.find(it => it.id === userUpdated.id).roles = userUpdated.roles;
+      return prev;
+    });
+  };
+
+  onSearchChange = (search) => {
+    this.tableHandler(0, this.state.sizePerPage, search, this.state.sort);
+  };
+
+  onPageChange = (page) => {
+    const offset = (page - 1) * this.state.sizePerPage;
+    this.tableHandler(offset, this.state.sizePerPage, this.state.search, this.state.sort, page);
+  };
+
+  onSizePerPageListHandler = (size) => {
+    this.setState(prev => {
+      prev.query.length = size;
+      return prev;
+    }, () =>{
+      this.tableHandler(this.state.query);
+    });
+  };
+
+  onSortChange = (sortName, sortOrder) => {
+    const sort = {
+      sortDirection: sortOrder,
+      orderColumn: SORT_NAME_INDEX[sortName]
+    };
+    this.tableHandler(0, this.state.sizePerPage, null, sort)
+  };
+
+
+  tableHandler = (offset, limit, search, sort, page) => {
+    let query = {
+        draw: 1,
+        start: offset,
+        length: limit,
+        orderColumn: sort.orderColumn,
+        sortDirection: sort.sortDirection,
+        searchValue: search,
+    };
+
+    User.getAllUsers(component.serverURL, query).then(result => {
+      const lastPage = Math.ceil(result.data.recordsFiltered / query.length);
+      this.setState(prev => {
+        prev.lastPage = lastPage;
+        prev.currentPage = page;
+        prev.isAdmin = this.state.isAdmin;
+        prev.users = result.data.data;
+        prev.recordsTotal = result.data.recordsTotal;
+        prev.recordsFiltered = result.data.recordsFiltered;
+        prev.sizePerPage = query.length;
+        prev.search = query.searchValue;
+        prev.sort = {
+          orderColumn : query.orderColumn,
+          sortDirection: query.sortDirection
+        };
+        return prev;
+      }, () => spinnerService.hideAll())
+    }).catch(error => {
+      this.setState(() => { throw error });
+    });
+  };
+
+   render() {
+    return(
+      div({ className: "roles-management" },[
+        span({ style: styles.pageTitle}, ["Roles Management"]),
+        Table({
+          headers: tableHeaders,
+          isAdmin: this.state.isAdmin,
+          data: this.state.users,
+          serverURL: component.serverURL,
+          editRole: this.editRoleHandler,
+          reviewFlow: true,
+          pagination: false,
+          tableHandler: this.tableHandler,
+          onSearchChange: this.onSearchChange,
+          onSortChange: this.onSortChange
+        }),
+        TablePaginator({
+          onPageChange: this.onPageChange,
+          currentPage: this.state.currentPage,
+          lastPage: this.state.lastPage
+        }),
+        RoleManagementEdit({
+          serverURL: component.serverURL,
```

Comment:
Ditto ... `RoleManagementEdit` should know about the serverURL regardless of where it is instantiated, so probably better off not as a prop.